types/bull/index.d.ts

```diff
@@ -228,6 +228,25 @@ declare namespace Bull {
      * Takes a lock for this job so that no other queue worker can process it at the same time.
      */
     takeLock(): Promise<number | false>;
+
+    /**
+     * Get job properties as Json Object
+     */
+    toJSON(): {
+      id: JobId,
+      name: string,
+      data: T,
+      opts: JobOptions,
+      progress: number,
+      delay: number,
+      timestamp: number,
+      attemptsMade: number,
+      failedReason: string,
```

Comment:
Where do you get that is always `string`?

Reply:
https://github.com/OptimalBits/bull/blob/bbae31dc585e7af019fa244518c71587ca5b7434/lib/job.js#L312

Every other place, like 

https://github.com/OptimalBits/bull/blob/bbae31dc585e7af019fa244518c71587ca5b7434/lib/job.js#L213
and 
https://github.com/OptimalBits/bull/blob/bbae31dc585e7af019fa244518c71587ca5b7434/lib/job.js#L424

To be safe, it could also be `null`.

> I'm not sure if `toJSON` is intended for public use.

See

https://github.com/OptimalBits/bull/blob/bbae31dc585e7af019fa244518c71587ca5b7434/lib/job.js#L497-L499

Reply:
But it could also be `undefined`. See constructor: https://github.com/OptimalBits/bull/blob/master/lib/job.js#L24

Reply:
i changed failedReason to any, could be null, undefined or even Object indeed.

> I'm not sure if `toJSON` is intended for public use.
Most private methods seem to be prefixed with an '_' and this method isn't in under the private method bloc.